api/v1/api_test.go

```diff
@@ -82,3 +114,83 @@ func TestStatusAPI(t *testing.T) {
 		t.Errorf("Wanted following build info %v, got %v.", testBuildInfo, responseBuildInfo)
 	}
 }
+
+func TestMetricsAPI(t *testing.T) {
+	dms := storage.NewDiskMetricStore("", 100*time.Millisecond, nil, logger)
+	TestApi := New(logger, dms, testFlags, testBuildInfo)
+
+	req, err := http.NewRequest("GET", "http://example.org/", &bytes.Buffer{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	w := httptest.NewRecorder()
+
+	testResponse := response{}
+	TestApi.metrics(w, req)
+	json.Unmarshal(w.Body.Bytes(), &testResponse)
```

Comment:
Wouldn't it be easier to provide a golden raw JSON string and compare it directly rather than unmarshalling and comparing the pieces?